portalnet/src/metrics/overlay.rs

```diff
@@ -1,81 +1,84 @@
-            "trin_utp_outcome",
-            "track success rate for all utp transfers outbound and inbound"
-        );
-        let utp_outcome_count_labels = &["protocol", "direction", "outcome"];
-        let utp_outcome_count = OverlayMetrics::register_counter_metric(
-            utp_outcome_count_options,
-            utp_outcome_count_labels,
-        );
-
-        let utp_active_count_options = opts!(
-            "trin_utp_active",
-            "count all active utp transfers outbound and inbound"
-        );
-        let utp_active_count_labels = &["protocol", "direction"];
-        let utp_active_count = OverlayMetrics::register_gauge_metric(
-            utp_active_count_options,
-            utp_active_count_labels,
-        );
-
-        let validation_count_options = opts!(
-            "trin_validation_total",
-            "count all content validations successful and failed"
-        );
-        let validation_count_labels = &["protocol", "success"];
-        let validation_count = OverlayMetrics::register_counter_metric(
-            validation_count_options,
-            validation_count_labels,
-        );
-
-        Self {
-            protocol: protocol.into(),
+    pub fn new(registry: &Registry) -> anyhow::Result<Self> {
+        let message_count = register_int_counter_vec_with_registry!(
+            opts!(
+                "trin_message_count",
+                "count all network messages sent and received"
+            ),
+            &["protocol", "direction", "type"],
+            registry
+        )?;
+        let utp_outcome_count = register_int_counter_vec_with_registry!(
+            opts!(
+                "trin_utp_outcome",
+                "track success rate for all utp transfers outbound and inbound"
+            ),
+            &["protocol", "direction", "outcome"],
+            registry
+        )?;
+        let utp_active_gauge = register_int_gauge_vec_with_registry!(
+            opts!(
+                "trin_utp_active",
+                "count all active utp transfers outbound and inbound"
+            ),
+            &["protocol", "direction"],
+            registry
+        )?;
+        let validation_count = register_int_counter_vec_with_registry!(
+            opts!(
+                "trin_validation_total",
+                "count all content validations successful and failed"
+            ),
+            &["protocol", "success"],
+            registry
+        )?;
+        Ok(Self {
             message_count,
             utp_outcome_count,
-            utp_active_count,
+            utp_active_gauge,
             validation_count,
-        }
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct OverlayMetricsReporter {
+    pub protocol: String,
+    pub overlay_metrics: OverlayMetrics,
+}
+
+impl OverlayMetricsReporter {
+    pub fn message_total(&self, direction: &str, message_type: &str) {
+        let protocol: &str = &self.protocol.to_string();
```

Comment:
`protocol` is already a `String`, it seems like `to_string()` is doing an extra memory allocation with no benefit. (and repeated below)